### DISCUSSION TITLE
Secure netmaker

### DISCUSSION THREAD

**Discussion by odedva** [gravitl/netmaker#3076] [Q&A] [answered]
Hi :)
we are considering netmaker open source deployment in our AWS project and would like to create ec2 instances so they will only be publicly accessible via 443\80 OR any other port via our VPC, currently netmaker request also 3479, 8089 (TCP), 51821 - 51830 (UDP).
are these are mandatory to be publicly open? is there a workaround around it ?

**Comment by abhishek9686** [comment, marked as answer]
Hi @odedva,
Yes, there is a workaround, by default the netmaker netclient uses ports in the range of 51821-51830 (UDP & TCP), but you can always change the listen port to your requirement, you can make clients listen on port 443 as well.
For more information on firewall requirements you can visit our docs https://docs.v2.netmaker.io/guide/getting-started/install#prerequisites
